packages/backend/src/plugins/explore.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createRouter,
+  getExampleTools,
+} from '@backstage/plugin-explore-backend';
+import { Router } from 'express';
+import { PluginEnvironment } from '../types';
+
+export default async function createPlugin(
+  env: PluginEnvironment,
+): Promise<Router> {
+  return await createRouter({
+    logger: env.logger,
+    toolProvider: {
+      getTools: getExampleTools,
```

Comment:
I wonder, should we just ship this with a pattern where all tools are just defined directly inline? How do you plan to ship this yourself?

Reply:
If you follow the `getExample` tools it has some very quick and dirty filtering logic on top of static data. For ours I was thinking of initially using something similar but the data is loaded from a yaml file for easier changes.

That said, I kind of envisioned the `explore-backend` package could provide different stock implementations (e.g. `YamlExploreToolProvider` or just `StaticExploreProvider.create(data)`).

Reply:
Ah I was thinking with a bit of extra wrapping, like you say. For example `StaticExploreToolProvider.fromData(...)`. I was actually gonna suggest YAML but realized you'd loose the type checking with that. Still, I think it makes sense and you can throw a bit of validation in the `.fromData(...)` to make it fail very early.

Reply:
It could also be loaded from the `app-config.yaml` as well. My first prototype of custom data did this and used the `$yaml` config loader to different files. Not sure if this should be a suggested option, but it could provide an easy setup example that's modifiable by adopters.

Reply:
I kinda hate the idea of putting it in config but I don't know exactly why 😅. Was considering it too but I think there's just a big risk of deployment complexity and/or bloated config YAML. Especially if you consider a setup where all config is injected through for example k8s resources. With a standalone YAML you get it all bundled into the build.

Reply:
Agreed. It provides a simple solution, but doesn't scale well or provide a good practice. I'll give this a bit more thought as well.